src/components/Payroll/PayrollList/PayrollListPresentation.tsx

```diff
@@ -1,46 +1,53 @@
+import type { Payroll } from '@gusto/embedded-api/models/components/payroll'
+import type { PayScheduleList } from '@gusto/embedded-api/models/components/payschedulelist'
 import { DataView, Flex } from '@/components/Common'
 import { useComponentContext } from '@/contexts/ComponentAdapter/useComponentContext'
 
-//TODO: Use Speakeasy type
-interface PayrollItem {
-  payrollId: string
-}
 interface PayrollListPresentationProps {
-  onRunPayroll: ({ payrollId }: PayrollItem) => void
-  payrolls: PayrollItem[]
+  onRunPayroll: ({ payrollId }: { payrollId: NonNullable<Payroll['payrollUuid']> }) => void
+  payrolls: Payroll[]
+  paySchedules: PayScheduleList[]
 }
 export const PayrollListPresentation = ({
   onRunPayroll,
   payrolls,
+  paySchedules,
 }: PayrollListPresentationProps) => {
   const { Badge, Button, Text } = useComponentContext()
   return (
     <DataView
       columns={[
         {
-          render: () => (
+          render: ({ payPeriod }) => (
             <Flex flexDirection="column">
-              <Text>Jul 5 - Jul 18, 2025</Text>
-              <Text>Regular Payroll</Text>
+              <Text>
+                {payPeriod?.startDate} - {payPeriod?.endDate}
+              </Text>
+              <Text>
+                {paySchedules.find(schedule => schedule.uuid === payPeriod?.payScheduleUuid)
+                  ?.name ||
+                  paySchedules.find(schedule => schedule.uuid === payPeriod?.payScheduleUuid)
+                    ?.customName}
+              </Text>
             </Flex>
           ),
           title: 'Pay period',
```

Comment:
i18n? Totally fine if we're waiting until later to do it all

Reply:
Probably faster to do it now than to ticket it 🤷🏻